src/OpenTK/Input/GamePadConfigurationSource.cs

```diff
@@ -32,22 +32,23 @@ namespace OpenTK.Input
     struct GamePadConfigurationSource
     {
         int? map_button;
-        JoystickAxis? map_axis;
+        int? map_axis;
         JoystickHat? map_hat;
         HatPosition? map_hat_position;
 
-        public GamePadConfigurationSource(JoystickAxis axis)
+        public GamePadConfigurationSource(bool axis, int index)
```

Comment:
I might prefer static factory-ish methods - GamePadConfigurationSource.CreateForAxis/Button/etc. Thoughts on that?